presto-main/src/main/java/com/facebook/presto/server/protocol/ExecutingStatementResource.java

```diff
@@ -98,6 +99,7 @@ public void getQueryResults(
             @QueryParam("maxWait") Duration maxWait,
             @QueryParam("targetResultSize") DataSize targetResultSize,
             @HeaderParam(X_FORWARDED_PROTO) String proto,
+            @HeaderParam(PrestoHeaders.PRESTO_PREFIX_URI) String xPrestoPrefixUri,
```

Comment:
I think the prefix URI should be itself a valid URI.  Can you validate that it's a proper URI?

Reply:
PRESTO_PREFIX_URI could be imported as X_FORWARDED_PROTO to keep the same way of using header references.

Reply:
> I think the prefix URI should be itself a valid URI. Can you validate that it's a proper URI?

The method QueryResourceUtil.prependUri validates that the concatenated string (prefixURI + Base64(coordinatorURI)) is a valid URI. If it's not valid an error is logged and the URIs are not modified.
We have to go through this validation when building the URI object.

Having said that, what would be the value of validating prefixURI on its own? There is a cost in terms of additional error handling/complexity and we wouldn't be catching any more errors I believe.

Reply:
The value of validating the prefix URI is that we can throw an error to the client that clearly explains why the request failed.  Using the present code, we silently (except for the log line) fall back to the original URI.  Ideally, we should let the client know exactly what was wrong with that prefix URI, and do not silently fallback, but rather explicitly fail the request with an HTTP 400 error (bad request).